app/controllers/dashboard_controller.rb

```diff
@@ -3,7 +3,31 @@ class DashboardController < ApplicationController
 
   def show
 
-    @projects = Project.all.where(user_id: current_user.id)
+    if user_details_complete == false
+      redirect_to user_details_path
+    else
+      @projects = Project.all.where(user_id: current_user.id)
+    end
+
+  end
+
+  private
+  def user_details_complete
+
+    user_details_fields_presence = []
+
+    user_details_fields_presence.push(current_user.name.present?)
+    user_details_fields_presence.push(current_user.date_of_birth.present?)
+    user_details_fields_presence.push(
+        (
+        current_user.line1.present? ||
+            current_user.townCity.present? ||
+            current_user.county.present? ||
+            current_user.postcode.present?
+        )
+    )
+
+    return user_details_fields_presence.include?(false) ? false : true
```

Comment:
```suggestion
    return user_details_fields_presence.all?
```
I think this is a more concise way to test for truthyness.